### PR TITLE
Implement Avatar component

### LINKED ISSUE #1345: Implement Avatar component
# 🙋 Feature Request

An avatar is a thumbnail representation of an entity, such as a user or an organization.

Screen Shot 2020-12-03 at 4 53 11 PM

## 💁 Possible Solution

The API should be something like the following.

```ts
interface SpectrumAvatarProps extends Omit<StyleProps, 'width' | 'height'> {
  src: string,
  alt?: string,
  /**
   * This affect both height and width at the same time.
   * @default size-100
  */
  size?: DimensionValue,
  isDisabled?: boolean
}
```

Example usages:

```tsx
<Avatar src="images/daniel.jpg" alt="Daniel" />
```

### Aria/Stately hooks
There are no hooks needed.

### DOM structure
An avatar should be implemented with the `img` tag. 
```tsx
<img src="images/daniel.jpg" alt="daniel"  />
```

## 🧢 Your Company/Team

RSP

### PR DESCRIPTION
**Pull request by solimant**
Closes #1345 

## ✅ Pull Request Checklist:

- [X] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [X] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [X] Filled out test instructions.
- [X] Updated documentation (if it already exists for this component).
- [X] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:
1. Open the `Avatar` storybook
2. In any of the stories (except the "disabled" one), inspect that `Avatar` can receive focus and show a focus ring

image

## 🧢 Your Project:
Adobe/RSP

### CODE REVIEW

File: packages/@adobe/react-spectrum/package.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
     "@react-aria/ssr": "^3.0.2",
     "@react-aria/visually-hidden": "^3.2.2",
     "@react-spectrum/actiongroup": "^3.2.0",
+    "@react-spectrum/avatar": "3.0.0-alpha.0",
```
Comment (snowystinger) [inline packages/@adobe/react-spectrum/package.json:26]: ah, no alphas in here

File: packages/@adobe/react-spectrum/src/index.ts [hunk 2]
```diff
@@ -11,6 +11,7 @@
  */
 
 export {ActionGroup} from '@react-spectrum/actiongroup';
+export {Avatar} from '@react-spectrum/avatar';
```
Comment (snowystinger) [inline packages/@adobe/react-spectrum/src/index.ts:14]: same thing

File: packages/@adobe/spectrum-css-temp/vars/spectrum-large.css [hunk 3]
```diff
@@ -71,6 +71,9 @@
   --spectrum-global-dimension-font-size-1100: 55px;
   --spectrum-global-dimension-font-size-1200: 62px;
   --spectrum-global-dimension-font-size-1300: 70px;
+  --spectrum-alias-avatar-size-100: 26px;
+  --spectrum-alias-avatar-size-400: 36px;
+  --spectrum-alias-avatar-size-600: 46px;
```
Comment (snowystinger) [inline packages/@adobe/spectrum-css-temp/vars/spectrum-large.css:76]: these don't have global-dimension values?

File: packages/@adobe/spectrum-css-temp/vars/spectrum-global.css [hunk 4]
```diff
@@ -451,6 +451,12 @@
   --spectrum-alias-heading-xxs-text-size: var(--spectrum-global-dimension-font-size-100);
   --spectrum-alias-heading6-margin-top: var(--spectrum-global-dimension-font-size-75);
   --spectrum-alias-heading-xxs-margin-top: var(--spectrum-global-dimension-font-size-75);
+  --spectrum-alias-avatar-size-50: var(--spectrum-global-dimension-size-200);
```
Comment (snowystinger) [inline packages/@adobe/spectrum-css-temp/vars/spectrum-global.css:454]: how'd you decide which ones went in which file?

File: packages/@react-spectrum/avatar/chromatic/Avatar.chromatic.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+ */
+
+import {Avatar} from '../';
+import React from 'react';
+import {storiesOf} from '@storybook/react';
+
+const SRC_URL_1 = 'https://mir-s3-cdn-cf.behance.net/project_modules/disp/690bc6105945313.5f84bfc9de488.png';
+const SRC_URL_2 = 'https://i.imgur.com/jjiXMsdb.jpg';
+
+storiesOf('Avatar', module)
```
Comment (snowystinger) [inline packages/@react-spectrum/avatar/chromatic/Avatar.chromatic.tsx:20]: For new stories/chromatic, we try to use CSF, there are some examples in the other chromatic stories
(sorry, plop hasn't been updated yet)

File: packages/@react-spectrum/avatar/docs/Avatar.mdx [hunk 6]
```diff
@@ -0,0 +1,65 @@
+<!-- Copyright 2021 Adobe. All rights reserved.
```
Comment (snowystinger) [inline packages/@react-spectrum/avatar/docs/Avatar.mdx:1]: just a heads up but all docs will need to be pulled out into another PR eventually, up to you when you do it

File: packages/@react-spectrum/avatar/package.json [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/adobe/react-spectrum"
+  },
+  "dependencies": {
+    "@react-aria/focus": "^3.4.0",
+    "@react-aria/utils": "^3.8.1",
+    "@react-spectrum/utils": "^3.6.1",
+    "@react-stately/collections": "^3.3.2",
```
Comment (snowystinger) [inline packages/@react-spectrum/avatar/package.json:37]: what're collections used for?

File: packages/@react-spectrum/avatar/src/Avatar.tsx [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+          {
+            'is-disabled': isDisabled
+          },
+          styleProps.className)}
+        src={src}
+        style={{
+          ...styleProps.style,
+          ...(sizeValue && {height: sizeValue, width: sizeValue})
+        }}
+        tabIndex={isDisabled ? undefined : 0} />
```
Comment (snowystinger) [inline packages/@react-spectrum/avatar/src/Avatar.tsx:57]: we'll want an aria package to contain the tabIndex/isDisabled relationship
we should probably contain the alt there as well
not sure if labels will come into play

File: packages/@react-spectrum/avatar/docs/Avatar.mdx [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+## Props
+
+<PropTable component={docs.exports.Avatar} links={docs.links} />
+
+## Visual options
+
+### Size
+
+```tsx example
+<Avatar alt="Zoe Saldana" size="size-1000" src="https://i.imgur.com/jjiXMsdb.jpg" />
```
Comment (snowystinger) [inline packages/@react-spectrum/avatar/docs/Avatar.mdx:53]: I don't think we can use her likeness in our docs, just a guess
we should use the same ones that spectrum.adobe.com has

File: packages/@react-types/avatar/src/index.d.ts [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+  /**
+   * Text description of the avatar.
+   *
+   * @default null
+   */
+   alt?: string,
+   /**
+    * The image URL for the avatar.
+    */
+   src: string
```
Comment (snowystinger) [inline packages/@react-types/avatar/src/index.d.ts:25]: technically this is optional https://spectrum.adobe.com/page/avatar/#-Use-generic-avatars-when-image-is-undefined

File: packages/@react-spectrum/avatar/src/Avatar.tsx [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+  } = useProviderProps(props);
+
+  const {styleProps} = useStyleProps(otherProps);
+
+  const sizeValue = size
+    ? dimensionValue(size)
+    : undefined;
+
+  return (
+    <FocusRing focusRingClass={classNames(styles, 'focus-ring')}>
```
Comment (snowystinger) [inline packages/@react-spectrum/avatar/src/Avatar.tsx:40]: if it's focusable, is it also pressable?

**Review by snowystinger — COMMENTED** [review]
thanks for getting this going!

**Review by solimant — COMMENTED** [review]
@snowystinger - please have another look

File: packages/@adobe/spectrum-css-temp/vars/spectrum-global.css (hunk 4 above)
Comment (snowystinger) [reply packages/@adobe/spectrum-css-temp/vars/spectrum-global.css:454]: gotcha, it's the newer stuff, i do not understand how that stuff is organized, resolving

File: packages/@react-types/avatar/src/index.d.ts (hunk 10 above)
Comment (snowystinger) [reply packages/@react-types/avatar/src/index.d.ts]: Ah yep, we interpreted it two different ways. Let's get some more opinions.
Down sides of both:
- optional image src means we would need to embed some images by default or code it to a discoverable place that we control so it can't be changed without us knowing
- required img src means potential for people forgetting to implement the generic icon or getting it from the wrong place

File: packages/@react-aria/avatar/test/useAvatar.test.js [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+      render(<ExampleAvatar onPress={mockOnPress} />);
+
+      const avatarElem = screen.getByRole('img');
+      expect(avatarElem).toHaveAttribute('tabIndex', '0');
+
+      userEvent.click(avatarElem);
+      expect(mockOnPress).toHaveBeenCalled();
+    });
+
+    it('calls and warns when onClick is used', () => {
```
Comment (snowystinger) [inline packages/@react-aria/avatar/test/useAvatar.test.js:70]: v2 avatar's never officially had onClick, so we shouldn't need to allow for it. We have it on button because it was something new and different compared to previous usages. This is now common for our library so we shouldn't need to allow it anymore

File: packages/@react-spectrum/avatar/src/Avatar.tsx [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+import {useAvatar} from '@react-aria/avatar';
+import {useProviderProps} from '@react-spectrum/provider';
+
+/**
+ * An avatar is a thumbnail representation of an entity, such as a user or an organization.
+ */
+export function Avatar(props: SpectrumAvatarProps) {
+  const {
+    isDisabled,
+    size = 'avatar-size-100',
```
Comment (snowystinger) [inline packages/@react-spectrum/avatar/src/Avatar.tsx:27]: I think we can shorten these to their size alone and omit the `avatar-size-` prefix
https://spectrum.adobe.com/page/avatar/#Table-of-options
I could be persuaded to keep the `size-` part since that's how other dimension values are entered in props

Comment (snowystinger) [reply packages/@react-spectrum/avatar/src/Avatar.tsx:27]: actually, do avatar sizes match size?

File: packages/@react-spectrum/avatar/src/Avatar.tsx [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+    src,
+    ...otherProps
+  } = useProviderProps(props);
+
+  const ref = useRef();
+
+  const {avatarProps} = useAvatar(props, ref);
+  const {alt, ...otherAvatarProps} = avatarProps;
+
+  const {styleProps} = useStyleProps(otherProps);
```
Comment (snowystinger) [inline packages/@react-spectrum/avatar/src/Avatar.tsx:37]: move this above useAvatar, this is usually one of the first hooks we call

File: packages/@react-spectrum/avatar/src/Avatar.tsx [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+
+  const sizeValue = size
+    ? dimensionValue(size)
+    : undefined;
+
+  return (
+    <FocusRing focusRingClass={classNames(styles, 'focus-ring')}>
+      <img
```
Comment (snowystinger) [inline packages/@react-spectrum/avatar/src/Avatar.tsx:45]: what happened to elementType?
https://github.com/adobe/react-spectrum/pull/2104/files#diff-6877c59b7583969ba245039e942bd0a915a3757410b8b2df2e6e56fb3d4360e1R45

Comment (snowystinger) [reply packages/@react-spectrum/avatar/src/Avatar.tsx:45]: can you add some stories that use other elementTypes? can it be svg as well as div? when would it be a div?

File: packages/@react-spectrum/avatar/stories/Avatar.stories.tsx [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+export const Disabled = AvatarTemplate.bind({});
+Disabled.args = {isDisabled: true, src: SRC_URL_1};
+Disabled.storyName = 'isDisabled';
+
+export const WithAltText = AvatarTemplate.bind({});
+WithAltText.args = {alt: 'Pensive', src: SRC_URL_2};
+WithAltText.storyName = 'with alt text';
+
+export const CustomSize = AvatarTemplate.bind({});
+CustomSize.args = {...WithAltText.args, size: 'size-1000'};
```
Comment (snowystinger) [inline packages/@react-spectrum/avatar/stories/Avatar.stories.tsx:46]: well now i'm confused, is it `avatar-size-1000` or `size-1000`
after looking at the spectrum.adobe.com more, I do see custom sizes, which you are correct should accept dimension values
we'll need to adjust the type for it so that it auto populates type ahead with the avatar sizes, I assume those are the preferential sizes

File: packages/@react-aria/avatar/src/useAvatar.ts [hunk 17]
```diff
@@ -0,0 +1,84 @@
...
+    ...(elementType !== 'img' && {
+      'aria-label': alt,
+      role: 'img'
+    }),
+    tabIndex: !isDisabled ? 0 : undefined
+  };
+
+  const domProps = filterDOMProps(otherProps);
+
+  const {isPressed, pressProps} = usePress({onPress, onPressStart, onPressEnd, isDisabled, ref});
```
Comment (LFDanLu) [inline packages/@react-aria/avatar/src/useAvatar.ts:67]: Mentioning this for other opinions:
I know in a previous PR that we made a stance that we only wanted to include isPressed and other states in the aria hook if absolutely necessary. What do we think about having it here? IMO I think we might be able to drop it here

File: packages/@react-spectrum/avatar/chromatic/Avatar.chromatic.tsx [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+Default.args = {src: SRC_URL_1};
+Default.storyName = 'default';
+
+export const Disabled = AvatarTemplate.bind({});
+Disabled.args = {isDisabled: true, src: SRC_URL_1};
+Disabled.storyName = 'isDisabled';
+
+export const WithAltText = AvatarTemplate.bind({});
+WithAltText.args = {alt: 'Pensive', src: SRC_URL_2};
+WithAltText.storyName = 'with alt text';
```
Comment (LFDanLu) [inline packages/@react-spectrum/avatar/chromatic/Avatar.chromatic.tsx:42]: nit: IMO we could drop this since alt text doesn't modify the look of the component right?

File: packages/@react-spectrum/avatar/src/Avatar.tsx (hunk 13 above)
Comment (LFDanLu) [reply packages/@react-spectrum/avatar/src/Avatar.tsx:27]: +1 to this. Judging from the changes made the global/med/large css files, avatar sizes != spectrum sizes (e.g. `avatar-size-700` = `size-500`). IMO it would easier if we simply redefined the size range provided by Spectrum to be consistent with the standard global sizes (e.g. the valid size value we accept are `size-200` to `size-500`).

File: packages/@react-spectrum/avatar/src/Avatar.tsx [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+import React, {useRef} from 'react';
+import {SpectrumAvatarProps} from '@react-types/avatar';
+import styles from '@adobe/spectrum-css-temp/components/avatar/vars.css';
+import {useAvatar} from '@react-aria/avatar';
+import {useProviderProps} from '@react-spectrum/provider';
+
+/**
+ * An avatar is a thumbnail representation of an entity, such as a user or an organization.
+ */
+export function Avatar(props: SpectrumAvatarProps) {
```
Comment (LFDanLu) [inline packages/@react-spectrum/avatar/src/Avatar.tsx:24]: Should this accept a ref argument?

File: packages/@react-spectrum/avatar/src/Avatar.tsx [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+    isDisabled,
+    size = 'avatar-size-100',
+    src,
+    ...otherProps
+  } = useProviderProps(props);
+
+  const ref = useRef();
+
+  const {avatarProps} = useAvatar(props, ref);
+  const {alt, ...otherAvatarProps} = avatarProps;
```
Comment (LFDanLu) [inline packages/@react-spectrum/avatar/src/Avatar.tsx:35]: Does `alt` need to be separated here? Can't you just spread `avatarProps` on img?

File: packages/@react-spectrum/avatar/src/Avatar.tsx (hunk 15 above)
Comment (LFDanLu) [reply packages/@react-spectrum/avatar/src/Avatar.tsx:45]: Do we wanna support other element types? The original issue seems to indicate that it should always be a `img`: https://github.com/adobe/react-spectrum/issues/1345

File: packages/@react-spectrum/avatar/stories/Avatar.stories.tsx [hunk 21]
```diff
@@ -0,0 +1,51 @@
...
+CustomSize.storyName = 'with custom size';
+
+export const Pressable = AvatarTemplate.bind({});
+Pressable.args = {...CustomSize.args, onPress: action('pressed')};
```
Comment (LFDanLu) [inline packages/@react-spectrum/avatar/stories/Avatar.stories.tsx:50]: Mind adding pressStart and pressEnd here as well?

File: packages/@react-spectrum/avatar/test/Avatar.test.js [hunk 22]
```diff
@@ -0,0 +1,61 @@
...
+    render(<Avatar src="http://localhost/some_image.png" isHidden />);
+    expect(screen.getByRole('img', {hidden: true})).toBeInTheDocument();
+  });
+
+  it('supports custom DOM props', () => {
+    render(<Avatar src="http://localhost/some_image.png" data-testid="Test avatar" />);
+    expect(screen.getByTestId(/test avatar/i)).toBeInTheDocument();
+  });
+
+  describe('when isDisabled = true', () => {
```
Comment (LFDanLu) [inline packages/@react-spectrum/avatar/test/Avatar.test.js:48]: nit: Would be nice to test that press event don't fire/fire depending on if it is disabled or not. Technically handled in the aria tests though heh so not a must do

**Review by LFDanLu — CHANGES_REQUESTED** [review]
Looks good for the most part, just some comments

**Review by solimant — COMMENTED** [review]
@snowystinger, @LFDanLu - thank you for the review. I've addressed your comments/requests. Happy to address any further asks.

File: packages/@react-spectrum/avatar/stories/Avatar.stories.tsx (hunk 16 above)
Comment (solimant) [reply packages/@react-spectrum/avatar/stories/Avatar.stories.tsx:45]: Heads up: updated thread link above ☝🏼

File: packages/@react-spectrum/avatar/src/Avatar.tsx (hunk 13 above)
Comment (snowystinger) [reply packages/@react-spectrum/avatar/src/Avatar.tsx]: hah, unlikely that we're ok with a 1px difference, i think we'll need to bring this up with design and find out how to communicate this best

Comment (snowystinger) [reply packages/@react-spectrum/avatar/src/Avatar.tsx]: i did just notice that this is a `size` prop, not `width`, so this may be less of an issue
though maybe we need to remove width/height as props that Avatar accepts?

File: packages/@react-spectrum/avatar/src/Avatar.tsx (hunk 15 above)
Comment (snowystinger) [reply packages/@react-spectrum/avatar/src/Avatar.tsx]: seems fair, i could see someone wanting to support svg pretty quickly

File: packages/@react-spectrum/avatar/stories/Avatar.stories.tsx (hunk 16 above)
Comment (snowystinger) [reply packages/@react-spectrum/avatar/stories/Avatar.stories.tsx:45]: yep, resolving this one

File: packages/@react-spectrum/avatar/src/Avatar.tsx (hunk 20 above)
Comment (snowystinger) [reply packages/@react-spectrum/avatar/src/Avatar.tsx]: that makes me think that avatarProps doesn't have the right return type, it should probably specify that it always returns the alt even if it's just a pass through

File: packages/@react-spectrum/avatar/package.json [hunk 23]
```diff
@@ -0,0 +1,53 @@
...
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/adobe/react-spectrum"
+  },
+  "dependencies": {
+    "@babel/runtime": "^7.14.6",
+    "@react-aria/avatar": "3.0.0-alpha.0",
+    "@react-aria/interactions": "^3.5.0",
+    "@react-aria/utils": "^3.8.1",
+    "@react-spectrum/utils": "^3.6.1",
```
Comment (LFDanLu) [inline packages/@react-spectrum/avatar/package.json:39]: Think we can remove these now

File: packages/@react-aria/avatar/package.json [hunk 24]
```diff
@@ -0,0 +1,33 @@
...
+    "src"
+  ],
+  "sideEffects": false,
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/adobe/react-spectrum"
+  },
+  "dependencies": {
+    "@babel/runtime": "^7.6.2",
+    "@react-aria/interactions": "^3.5.0",
```
Comment (LFDanLu) [inline packages/@react-aria/avatar/package.json:22]: don't need "@react-aria/interactions" or "@react-types/shared" here either anymore

**Review by LFDanLu — CHANGES_REQUESTED** [review]
LGTM, just a couple `package.json` cleanup bits

**Review by snowystinger — COMMENTED** [review]
I got clarification from design, this is how the 'size' prop should behave

```tsx
<Avatar size="avatar-size-100" /> // allowed
<Avatar size="600px" /> // allowed
<Avatar size="30em" /> // allowed
<Avatar size="100" /> // NOT allowed
<Avatar size="size-100" /> // NOT allowed
<Avatar height="anything" /> // NOT allowed at all
<Avatar width="anything" /> // NOT allowed at all
```

**Review by solimant — COMMENTED** [review]
@LFDanLu - I've made the changes, please have another look.

> i did just notice that this is a size prop, not width, so this may be less of an issue
though maybe we need to remove width/height as props that Avatar accepts?

@snowystinger - yes, we're removing `width` and `height` in `SpectrumAvatarProps`.

> I got clarification from design, this is how the 'size' prop should behave

@snowystinger - thank you for checking. I've approached in a way that should satisfy that ask. I'm making an assumption, though, that disallowed values default to the default value; i.e. `avatar-size-100`. Please have another look.

File: packages/@react-aria/avatar/src/useAvatar.ts [hunk 25]
```diff
@@ -0,0 +1,60 @@
...
+export function useAvatar(props: AriaAvatarOptions): AvatarAria {
+  const {
+    elementType = 'img',
+    alt = '',
+    isDisabled,
+    ...otherProps
+  } = props;
+
+  const avatarProps = {
+    ...(elementType === 'img' && {alt}),
```
Comment (snowystinger) [inline packages/@react-aria/avatar/src/useAvatar.ts:47]: I think if it's not labelled, then it's supposed to be aria-role="presentation"
@majornista ?
This is the dom in the default story
Screen Shot 2021-07-31 at 1 35 48 PM

File: packages/@react-aria/avatar/src/useAvatar.ts [hunk 26]
```diff
@@ -0,0 +1,60 @@
...
+    ...(elementType !== 'img' && {
+      'aria-label': alt,
+      role: 'img'
+    }),
+    tabIndex: !isDisabled ? 0 : undefined
```
Comment (snowystinger) [inline packages/@react-aria/avatar/src/useAvatar.ts:52]: We need to add focus styles if we're going support it being focusable

File: packages/@react-spectrum/avatar/src/Avatar.tsx [hunk 27]
```diff
@@ -0,0 +1,66 @@
...
+    src,
+    ...otherProps
+  } = useProviderProps(props);
+
+  const {styleProps} = useStyleProps(otherProps);
+  const domRef = useDOMRef(ref);
+  const {avatarProps} = useAvatar(props);
+  const {alt, ...otherAvatarProps} = avatarProps; // alt for jsx-a11y/alt-text
+
+  const sizeValue = typeof size !== 'number' && (SIZE_RE.test(size) || !isNaN(size as any))
```
Comment (snowystinger) [inline packages/@react-spectrum/avatar/src/Avatar.tsx:37]: doesn't `typeof size !== 'number'` prevent `!isNaN(size)` from ever being true? or am i confused with double negatives?

File: packages/@react-aria/avatar/src/useAvatar.ts (hunk 25 above)
Comment (solimant) [reply packages/@react-aria/avatar/src/useAvatar.ts:46]: As far as screen readers are concerned, having `alt=""` is effectively the same as `role="presentation"`, but the latter is not widely supported:
> Screen readers also allow the use of WAI-ARIA to hide elements by using `role="presentation"`. However, currently, this feature is not as widely supported as using a null `alt` attribute.

Source: https://www.w3.org/WAI/tutorials/images/decorative/#image-used-as-part-of-page-design.

That said, and according to my understanding from the below [excerpt](https://www.w3.org/TR/wai-aria-1.1/#presentation) from WAI-ARIA, we'd want to add a `role="presentation"` if `img` is part of a containing element that provides the role and alternative text of the overall construct:
> Authors _**SHOULD NOT**_ provide meaningful alternative text (for example, use `alt=""` in HTML) when the `presentation` role is applied to an image.
> 
> In the following code sample, the containing `img` and is appropriately labeled by the caption paragraph. In this example the `img` element can be marked as presentation because the role and the text alternatives are provided by the containing element.
> 
> EXAMPLE 10
>  ```html
>  <div role="img" aria-labelledby="caption">
>    
>    <p id="caption">A visible text caption labeling the image.</p>
>  </div>
>  ```

Another thing is, since the `presentation` role causes a given element to be treated as having no role or to be removed from the accessibility tree, do want that for Avatar? It doesn't seem to me that we'd want to omit it from the accessibility tree. I personally see Avatar as being a functional/informative image vs being a decorative image, so if anything, having a prominent role for screen readers is more fitting IMO.

File: packages/@react-aria/avatar/src/useAvatar.ts (hunk 26 above)
Comment (solimant) [reply packages/@react-aria/avatar/src/useAvatar.ts]: Removed since we agreed that focusable/pressable is not happening now.

File: packages/@react-spectrum/avatar/src/Avatar.tsx (hunk 27 above)
Comment (solimant) [reply packages/@react-spectrum/avatar/src/Avatar.tsx:39]: I agree that this part is not super clear. Added a comment.
Basically, we want to cover the case where `size` is a `string` number, such as `"300"` (as opposed to a `number` number; e.g. `300`). In this case, `size` is not of type `number`, but `isNaN` should still return `false`:
```ts
isNaN(300); // false, because 300 is a number
isNaN('300'); // false, because '300' is, essentially, a number
isNaN('300px'); // true, because '300px' is not a number
```
Also, casting `size` as `any` here is a hack since `isNaN` expects `number`, but I want it to handle the `string` number.

**Review by solimant — COMMENTED** [review]
@snowystinger - I've addressed your comments. Please have a look. Thank you 🙏🏼

File: packages/@react-aria/avatar/src/useAvatar.ts (hunk 25 above)
Comment (snowystinger) [reply packages/@react-aria/avatar/src/useAvatar.ts:46]: fascinating, thanks for going the extra mile in your research on this one

File: packages/@react-spectrum/avatar/src/Avatar.tsx (hunk 27 above)
Comment (LFDanLu) [reply packages/@react-spectrum/avatar/src/Avatar.tsx:39]: Just wanna make sure: should passing `size="300"` work? Doesn't seem like it does locally

**Review by LFDanLu — APPROVED** [review]
LGTM, thanks!

File: packages/@react-aria/avatar/src/useAvatar.ts [hunk 28]
```diff
@@ -0,0 +1,58 @@
...
+   */
+  avatarProps: ImgHTMLAttributes<HTMLElement>
+}
+
+/**
+ * Provides the behavior and accessibility implementation for an avatar
+ * component. An avatar is a thumbnail representation of an entity, such as a
+ * user or an organization.
+ */
+export function useAvatar(props: AriaAvatarOptions): AvatarAria {
```
Comment (snowystinger) [inline packages/@react-aria/avatar/src/useAvatar.ts:38]: we can leave this here for now because the packages are still private and we may decide to move useAvatar into the Avatar component as it may be a very specific to Spectrum component

File: packages/@react-spectrum/avatar/chromatic/Avatar.chromatic.tsx [hunk 29]
```diff
@@ -0,0 +1,42 @@
...
+export const Default = AvatarTemplate.bind({});
+Default.args = {src: SRC_URL_1};
+Default.storyName = 'default';
+
+export const Disabled = AvatarTemplate.bind({});
+Disabled.args = {isDisabled: true, src: SRC_URL_1};
+Disabled.storyName = 'isDisabled';
+
+export const CustomSize = AvatarTemplate.bind({});
+CustomSize.args = {size: 'size-1000', src: SRC_URL_2};
```
Comment (snowystinger) [inline packages/@react-spectrum/avatar/chromatic/Avatar.chromatic.tsx:41]: This one is an invalid size now, it'll just do default
```suggestion
CustomSize.args = {size: 100, src: SRC_URL_2};
```

**Review by snowystinger — COMMENTED** [review]
One last clean up item and then I'm happy to approve as well

**Comment by solimant** [comment]
@snowystinger, @LFDanLu - thank you very much for the review and comments. I've addressed the requested changes, and have also introduced support for more element types via 16f710c. I'm happy to omit those changes and refactor whatever is necessary.

Also, on a similar note, and further to our offline conversations, I understand we want to repurpose `useAvatar` to be something more generic. Perhaps we can call it `useGraphic`, or something of the like..? This can then encapsulate all ARIA-related logic for anything that depends on graphical HTML elements; e.g. `svg`, `img`, etc., and could also be used in the `Image` component. If that makes sense, or if it's at least a good start, I'm happy to pull this into its own PR and refactor what's necessary here to close things off.

**Comment by LFDanLu** [comment]
cc @dannify regarding the above. We talked about this offline previously, do we want to go ahead and move the hook's contents into the Avatar component for now? IMO `useGraphic` might be a good name for this hook and we can split it out into another pull to discuss what else the hook may cover.

**Comment by dannify** [comment]
I think for the first release I would like to merge the hook into React Spectrum. I think we will need it down the track but we don't have enough of a use case to make a pattern out of it just yet.

**Comment by solimant** [comment]
Thank you, @dannify - consolidation is done and ready for review.

**Review by snowystinger — APPROVED** [review]
lgtm! thanks!

**Review by LFDanLu — APPROVED** [review]
LGTM, thanks!
